MultiServer.py

```diff
@@ -1802,14 +1803,46 @@ def _cmd_alias(self, player_name_then_alias_name):
             self.output(response)
             return False
 
+    def resolve_player(self, input_name: str) -> typing.Optional[typing.Tuple[int, int, str]]:
+        """ returns (team, slot, player name) """
+        # first match case
+        for (team, slot), name in self.ctx.player_names.items():
+            if name == input_name:
+                return team, slot, name
+        
+        @dataclass
+        class TSNC:
+            """ team, slot, name, (count without case) """
+            team: int
+            slot: int
+            name: str
+            count: int = 1
+
+        # if no case-sensitive match, then match without case only if there's only 1 match
+        lowered_data: typing.Dict[str, TSNC] = {}
+        for (team, slot), name in self.ctx.player_names.items():
+            lowered = name.lower()
+            if lowered in lowered_data:
+                lowered_data[lowered].count += 1
```

Comment:
i would instead do
```python
match: Optional[Tuple[int, int, str]] = None
for ...
    if match:
        return None  # ambiguous, abort.
    else:
        match = (team, slot, name)

return match
```
this breaks out faster and might be easier to follow. also it gets rid of the noisy dataclass default-count handling.
if the server should print "the name is ambiguous" as the reply to a command, i think that should be handled via an esception (instead of `return None`).